#    Description                  : stripPointerCastsAndOffsets should not process across AddrSpaceCast
#    When Fixed in Open Source    : Need Always for OCL
#    P4V CL                       : 552346(3.8.0), 642680(4.0.0)
#    Category                     : Bug
#    Author                       : jliu29
#    Open Source Classification   : Internal Only
#    Impact                       :
#    Failing Workload/Commandline :
#    Notes                        : I cannot upstream this one, since it exposes LLVM's other GAS issues,
#                                   including several unit tests. This patch is not a must-have, and IGC
#                                   can function even w/o this patch.

diff -Naur --strip-trailing-cr a/lib/IR/Value.cpp b/lib/IR/Value.cpp
--- a/lib/IR/Value.cpp	2017-10-24 10:42:55.731712686 -0400
+++ b/lib/IR/Value.cpp	2017-10-24 10:43:09.940712129 -0400
@@ -467,6 +467,11 @@
       V = GEP->getPointerOperand();
     } else if (Operator::getOpcode(V) == Instruction::BitCast ||
                Operator::getOpcode(V) == Instruction::AddrSpaceCast) {
+    // Do not process across AddrSpaceCast, since its result could
+    // be a nullptr depending on the meaning of src and dest addrspace.
+    if (Operator::getOpcode(V) == Instruction::AddrSpaceCast) {
+        return V;
+    }
       V = cast<Operator>(V)->getOperand(0);
     } else if (GlobalAlias *GA = dyn_cast<GlobalAlias>(V)) {
       if (StripKind == PSK_ZeroIndices || GA->isInterposable())
